### PR TITLE
Support per-child alignment

### LINKED ISSUE #19: Support equivalent of align-self

### PR DESCRIPTION
**Pull request by peterstuart**
This is equivalent to `align-self` from flexbox.

Fixes #19.

### CODE REVIEW

File: examples/container.rs [hunk 1]
```diff
@@ -18,8 +18,20 @@ fn main() -> Result<(), Infallible> {
     let mut display: SimulatorDisplay<Rgb888> = SimulatorDisplay::new(display_size);
 
     let character_style = MonoTextStyle::new(&FONT_10X20, Rgb888::BLACK);
-    let text1 = Text::new(TextOptions { character_style }, "Line 1");
-    let text2 = Text::new(TextOptions { character_style }, "Line 2");
+    let text1 = Text::new(
+        TextOptions {
+            character_style,
+            layout_options: Default::default(),
```
Comment (peterstuart) [inline examples/container.rs:24]: The ergonomics of this are bad, but I'm planning on redoing all the options stuff with a builder pattern anyway, so not worrying about it for now.

File: src/widget/container/mod.rs [hunk 2]
```diff
@@ -155,16 +160,21 @@ where
         };
 
         for child in &self.options.children {
-            let default_size = match (self.options.alignment, self.main_axis()) {
+            let child_alignment = child
+                .layout_options()
+                .alignment
+                .unwrap_or(self.options.alignment);
+
+            let default_size = match (child_alignment, self.main_axis()) {
```
Comment (peterstuart) [inline src/widget/container/mod.rs:168]: This is the only behavior change in the code, the rest is just moving things around. (Seems suspiciously easy?)

Comment (helixbass) [reply src/widget/container/mod.rs:168]: Makes sense conceptually that the child can "override" the parent by supplying its own configuration value

**Review by helixbass — APPROVED** [review]
@peterstuart looks good

**Pull request merged**